plugin.rb

```diff
@@ -27,59 +27,46 @@
 
 module ::DiscourseAi
   PLUGIN_NAME = "discourse-ai"
+  module NSFW
+  end
 end
 
-require_relative "lib/discourse_ai/engine"
+Rails.autoloaders.main.push_dir(File.join(__dir__, "lib"), namespace: ::DiscourseAi)
+# this inflection is finicky
+Rails.autoloaders.main.push_dir(File.join(__dir__, "lib/nsfw"), namespace: ::DiscourseAi::NSFW)
+
+require_relative "lib/engine"
 
 after_initialize do
-  require_relative "lib/shared/inference/discourse_classifier"
-  require_relative "lib/shared/inference/discourse_reranker"
-  require_relative "lib/shared/inference/openai_completions"
-  require_relative "lib/shared/inference/openai_embeddings"
-  require_relative "lib/shared/inference/openai_image_generator"
-  require_relative "lib/shared/inference/anthropic_completions"
-  require_relative "lib/shared/inference/stability_generator"
-  require_relative "lib/shared/inference/hugging_face_text_generation"
-  require_relative "lib/shared/inference/amazon_bedrock_inference"
-  require_relative "lib/shared/inference/cloudflare_workers_ai"
-  require_relative "lib/shared/inference/hugging_face_text_embeddings"
-  require_relative "lib/shared/inference/function"
-  require_relative "lib/shared/inference/function_list"
-
-  require_relative "lib/shared/classificator"
-  require_relative "lib/shared/post_classificator"
-  require_relative "lib/shared/chat_message_classificator"
-
-  require_relative "lib/shared/tokenizer/tokenizer"
-
-  require_relative "lib/shared/database/connection"
-
-  require_relative "lib/completions/entry_point"
-
-  require_relative "lib/modules/nsfw/entry_point"
-  require_relative "lib/modules/toxicity/entry_point"
-  require_relative "lib/modules/sentiment/entry_point"
-  require_relative "lib/modules/ai_helper/entry_point"
-  require_relative "lib/modules/embeddings/entry_point"
-  require_relative "lib/modules/summarization/entry_point"
-  require_relative "lib/modules/ai_bot/entry_point"
-  require_relative "lib/discourse_automation/llm_triage"
+  Rails.autoloaders.each do |autoloader|
+    autoloader.inflector.inflect({ "llm" => "LLM", "chat_gpt" => "ChatGPT" })
+  end
+
+  # do not autoload this cause we may have no namespace
+  require_relative "discourse_automation/llm_triage"
+
+  # jobs are special, they live in a discourse ::Jobs
+  require_relative "jobs/regular/create_ai_reply"
```

Comment:
Chat doesn't need this -- you should be able to put the jobs into `app/jobs/regular/discourse_ai/` (or `app/jobs/scheduled/discourse_ai/`) and they will autoload. Then their class path will be `Jobs::DiscourseAi::CreateAiReply` for example, and you won't need all this require_relative stuff.